### ISSUE TITLE
Deprecate `create_model`

### ISSUE THREAD

**Issue by francisco-dlp** [hyperspy/hyperspy#568]
I propose to turn the `hspy.create_model` function into a `Signal` method that should allow us to write:

```python
 m = s.create_mode()

```

Instead of 

```python

m = create_model(s)

```

The advantage is that, given that there are options which are specific for each `Signal` type, the docstring and argument list will be less confusing. What do you think?

**Comment by to266** [comment]
Sure.

Additionally, the way I created to save whole model will work way better
(it essentially creates a dictionary of the model and then puts in the metadata of the signal, which can now be saved. To load the model, you'd have to load the signal, create a model with a signal, and then call `load` on the model... that would get rid of the two last calls)

As a consequence, I guess we change the model to be inside the signal, not the other way around?

**Comment by francisco-dlp** [comment]
That would be nice but then how do we deal with EELS models that require two signals, the low-loss and the core-loss?

**Comment by to266** [comment]
Is the low-loss usually just a single spectrum, or just as big (and possibly multidimensional) as the core-loss? If just one, we can save it as well.

Additionally, I guess a person would want to work with both signals anyway, so combining them into a single package / signal would be useful - no need to copy two files, as you have everything about the measurement in the one combo-file ;)

**Comment by francisco-dlp** [comment]
The low-loss is normally a Signal of the same dimensions as the core-loss. What about saving the model itself and offering the option to save the `Signal`(s) path or the `Signal`(s) themselves at saving time? At loading time, if the loading routine doesn't find the `Signal`(s) in the stored path it could request the user to provide the right path(s) via arguments.

**Comment by to266** [comment]
I guess that could be done.... However I don't imagine myself ever using "store the path of the signal(s)" option. I implemented the saving of the model in the metadata due to easier integration with hdf5 - at the moment our structure lets save "a signal + anything in it's metadata". In order to save a model, a completely new structure would have to be created (or some sort of "root" signal would have to be hacked)

Also, in the end the signal is the main building block - a model without a signal makes little sense, so I'm not sure saving just a model is very convenient. Besides, the storage space now is of little importance (i.e. having big files is fine). If nothing else, I think the main problem is having too many files - so combining core-loss, low-loss and the model into a single file (instead of the 3) would benefit reducing the clutter / how to transport the data

**Comment by pburdet** [comment]
To be complete and easier to use, it would be nice to be able to save a signal containing one or more models and to be able to save a model of several signals.

**Comment by to266** [comment]
I agree with saving multiple models for a signal, but what do you mean by saving "a model of several signals", @pburdet?

**Comment by pburdet** [comment]
This might be for another Issue, but If you have several acquisition with the same experimental parameters, you might want the same model for all. Actually you could pass over the model from one signal to the other. It makes more sense in the case of decomposition.

**Comment by to266** [comment]
Ok, I see. What exactly, do you think, we should be able to transfer? I guess the obvious one is the components (e.g. the same collection of Gaussians + their relations (twins, fixed parameters)), but the actual values would change throughout (different intensities / widths / energies)?

Though to be able to just "clone structure" of the model might be useful, I agree. That should be a rather trivial change once the model -> dictionary branch is merged, though

**Comment by vidartf** [comment]
Does it make sense to save things as some sort of a project? It would either embed the data it needs (relevant signals and models), or link to them ("model A is found in this path, and it uses signal A and B; signal A is found here; signal B is found here (relative paths)"). If we saved like that, than any combination of models and signals can be saved as the internal relationships are stored.

**Comment by to266** [comment]
I think the original structure was intended to use like that, however I have not tested if it works.

I don't like saving paths to things though, I think it just creates more clutter. At least not by default (I suppose it would be useful when signals are very large, as in @pburdet case).

**Comment by magnunor** [comment]
I'm in favor of some kind of "project" structure for certain datasets. For example: a line scan containing several EELS core loss edges, lowloss signal, EDS, HAADF-signal, survey image, ... Currently I store them separately, which can create quite a bit of clutter. It would be very nice to have all these in one file, including one or several models. 

With regards to big files: has anyone tried the in-built compression in HDF5? On certain datasets it seems to help quite a bit: from 8 GB to 2.2.

**Comment by to266** [comment]
For project structure, I created a PR with the minimal implementation of them here: https://github.com/hyperspy/hyperspy/pull/576

Obviously it's still WIP, but please let me know if that works and makes sense / if you have any "feature requests"

**Comment by francisco-dlp** [comment]
Fixed in #624
